File: Examples.vb/Examples/EnumExamples.vb
Imports NUnit.Framework

Namespace Examples
	Public Class EnumExamples

		Enum ExampleEnum
			EnumOne
			EnumTwo
		End Enum

		Public Shared Function GetRandomEnum(Of T)() As T

			Dim v As T() = [Enum].GetValues(GetType(T))
			Return CType(v.GetValue(New Random().[Next](v.Length)), T)

		End Function

		<Test> Public Sub GetRandomEnum_Example()
			Assert.That(GetRandomEnum(Of ExampleEnum)(), [Is].TypeOf(GetType(ExampleEnum)))
		End Sub

		Public Shared Function GetEnumFromName(Of T)(ByVal name As String) As T

			Try
				Dim theEnum As T = CType([Enum].Parse(GetType(T), name), T)
				If [Enum].IsDefined(GetType(T), theEnum) Then
					Return theEnum
				End If
			Catch e As ArgumentException
			End Try
			Return Nothing

		End Function

		<Test()> Public Sub GetEnumFromName_Example()

			For Each item As EnumExamples.ExampleEnum In [Enum].GetValues(GetType(ExampleEnum))
				Dim statusFromString As ExampleEnum = GetEnumFromName(Of ExampleEnum)(item.ToString())
				Assert.That(statusFromString, [Is].EqualTo(item))
			Next

		End Sub

	End Class
End Namespace

File: Nunit.Test/Examples/System/EnumExamples.cs
using NUnit.Framework;
using System;

namespace Examples.System.Examples
{
	[TestFixture]
	public class EnumExamples
	{
		public enum TestEnum
		{
			Zero = 0,
			One = 1,
			Two = 2
		}

		[Test]
		public void LoopThroughEnum()
		{
			foreach (var testEnum in (TestEnum[])Enum.GetValues(typeof(TestEnum)))
			{
				//do something with the enum
				Assert.That(testEnum, Is.Not.Null);
			}
		}

		[TestCase(TestEnum.One, 1)]
		public void ConvertToInteger(TestEnum theEnum, int enumIntValue)
		{
			var convertedValue = Convert.ToInt32(theEnum);
			var castedValue = (int) theEnum;
			Assert.Multiple(() =>
			{
				Assert.That(convertedValue, Is.TypeOf<int>());
				Assert.That(convertedValue, Is.EqualTo(enumIntValue));
				Assert.That(castedValue, Is.EqualTo(convertedValue));
			});
		}

		[TestCase(TestEnum.One, "One")]
		public void EnumToString(TestEnum theEnum, string enumName)
		{
			Assert.Multiple(() =>
			{
				Assert.That(theEnum.ToString(), Is.EqualTo(enumName));
				Assert.That($"{theEnum}", Is.EqualTo(enumName));
			});
			
		}
	}
}
